dynamorm/model.py

```diff
@@ -459,3 +459,13 @@ def update(self, conditions=None, update_item_kwargs=None, **kwargs):
             setattr(self, key, val)
 
         return resp
+
+    def delete_item(self):
+        """Delete this record in the table."""
+        delete_item_kwargs = {self.Table.hash_key: getattr(self, self.Table.hash_key)}
+        try:
+            delete_item_kwargs[self.Table.range_key] = getattr(self, self.Table.range_key)
+        except (AttributeError, TypeError):
+            pass
```

Comment:
This whole block to update kwargs with the hash and range key is repeated above, let's pull it out into a private helper method so we can reuse it on both -- and anything else in the future.